frontend-student/src/components/courseDetails/CourseDetails.jsx

```
@@ -64,30 +128,44 @@ export default function CourseDetails() {
+              Contact: {course.instructor.contactNumber}
+            </p>
           </div>
 
-          <button className="w-full mt-6 bg-blue-500 text-white py-3 rounded-lg hover:bg-blue-600 transition">
-            Buy This Course
+          <button
+            className='w-full mt-6 bg-blue-500 text-white py-3 rounded-lg hover:bg-blue-600 transition disabled:bg-blue-300'
+            onClick={handleBuyCourse}
+            disabled={purchasing}
+          >
+            {purchasing ? 'Processing...' : 'Buy This Course'}
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)